packages/smooth_app/lib/data_models/continuous_scan_model.dart

```diff
@@ -35,10 +36,12 @@ class ContinuousScanModel with ChangeNotifier {
       ProductList(listType: ProductList.LIST_TYPE_SCAN, parameters: '');
 
   bool _contributionMode;
-  String _barcodeLatest;
+  String _barcodeLatestScan;
+  String _barcodeLatestSuccessfulScan;
```

Comment:
What exactly do we have both of them, is it something internal or have I missed any possible usecase

Reply:
The first is about the last barcode the scanner tool was successful with. When you scan, a callback with the same barcode is called several times in a second, which may make sense in the scanner point of view (each time, that's the barcode it has just scanned, and it's the same one 100ms later), but needs to be dealt with in our case (we need to look for that barcode only once, not every 100ms).

The second is about the scan process "on our side" after the barcode is scanned:
1. if we find the barcode in the database, we display it and have it refreshed from the server
2. if we cannot find it in the database, we directly try to fetch it from the server
In both cases there's only one "scan timestamp" to add, and most of the time a "refresh timestamp" to add.
The second variable is helpful not to consider that when you refresh the data from the server after finding data in the database, it's not another scan, and therefore there's no other "scan timestamp" to add.

I was a bit verbose, wasn't I?

I'll try to rename them today. If I cannot find appropriate names, I'll merge anyway - we're talking about private variables.